### PR TITLE
Fix some yarpmotorgui memory leaks and warnings

### PR DESCRIPTION
**Pull request by S-Dafarra**
It also changes the default messages shown in the HF tab to ``UNKNOWN``.

### CODE REVIEW

**Comment by S-Dafarra** [comment]
> Thanks for opening this pull request! The maintainers of this repository would appreciate it if you would update the release notes by adding a file in `doc/release/<target_branch>`, based on your changes.

Regarding this, any suggestion on where to document the changes? Should I create a new file in the ``doc/releases/master`` folder?

**Review by randaz81 — APPROVED** [review]

**Comment by randaz81** [comment]
> Thanks for opening this pull request! The maintainers of this repository would appreciate it if you would update the release notes by adding a file in `doc/release/<target_branch>`, based on your changes.

In this PR there are only bug fixes and no new features, so I think a changelog is not required. Thank you!

File: src/yarpmotorgui/partitem.h [hunk 1]
```diff
@@ -124,7 +124,7 @@ class PartItem : public QWidget
     bool    m_part_motorPositionVisible;
     bool    m_part_dutyVisible;
     bool    m_part_currentVisible;
-    yarp::dev::InteractionModeEnum* m_interactionModes;
```
Comment (drdanz) [inline src/yarpmotorgui/partitem.h:127]: The change is technically correct, but if you look at the whole file, you can see that there are several pointer-vectors and no `std::vector`. This is a change that, in my opinion, makes the code less readable.
I'd rather change all of them or just add the missing delete...

Comment (S-Dafarra) [reply src/yarpmotorgui/partitem.h]: Done in https://github.com/robotology/yarp/pull/2774/commits/601570e754d41796bc02931121d08268609a3934

**Review by drdanz — APPROVED** [review]
LGTM

**Comment by drdanz** [comment]
Merged, thanks.

**Comment by S-Dafarra** [comment]
I just noticed that after https://github.com/robotology/yarp/pull/2774/commits/d28d58e647995c2eff4043404d05fa4f016e8547, ``yarpmotorgui`` may crash when failing to open a part. This should be fixed with #2791

**Pull request merged**